lib/Net/Stripe.pm

```diff
@@ -198,9 +198,22 @@ Charges: {
                        Int :$application_fee?,
                        Str :$receipt_email?
                      ) {
+
+        my $customer_id;
+        if ( defined( $customer ) ) {
+            if ( ref( $customer ) eq 'Net::Stripe::Customer' ) {
+                $customer_id = $customer->id;
+            } elsif ( ref( $customer ) eq 'HASH' ) {
+                $customer = $self->post_customer( %$customer );
+                $customer_id = $customer->id;
```

Comment:
I always worry when I see a subroutines parameters being modified - `$customer` in this case.

How about this?
```
my $customer_obj = $self->post_customer( %$customer );
$customer_id = $customer_obj->id;
```